runtime/common/src/precompile/input.rs

```diff
@@ -186,6 +187,21 @@ where
 
 		Ok(bytes.to_vec())
 	}
+
+	fn bool_at(&self, index: usize) -> Result<bool, Self::Error> {
+		let param = self.u64_at(index)?;
```

Comment:
`self.u256_at(index)
			.map(|x| !x.is_zero())
			.map_err(|_| PrecompileFailure::Revert {
				exit_status: ExitRevert::Reverted,
				output: "failed to decode bool".into(),
				cost: self.target_gas.unwrap_or_default(),
			})`

Reply:
Yeah I thought about that too, but then a random integer value will be true rather than erroring out. 

But maybe this behavior is fine too, I don't have a strong opinion

Reply:
the call will come from predeployed contract and will be encoded so it will either be one or zero